03week/towersOfHanoi.js

```diff
@@ -19,30 +19,58 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  console.log("I am inside move piece", startStack, endStack);
+  var movedPiece = stacks[startStack].pop();
+  return stacks[endStack].push(movedPiece);
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+  console.log("I am inside the is legal method");
+  var startStackValues = stacks[startStack];
+  var endStackValues = stacks[endStack];
+  var endLength = endStackValues.length;
+  if (endLength == 0) {
+    return true;
+  } else if (startStack.length == 0) {
+    return false;
+  }
+  //end/stack is being sliced before the word starts to the first
+  if (endStackValues.slice(-1)[0] > startStackValues.slice(-1)[0]) {
+    return true;
+  } else {
+    return false;
+  }
 }
-
-function checkForWin() {
-  // Your code here
-
+function checkForWin(startStack, endStack) {
+  if (stacks.c.length == 4 || stacks.b.length == 4) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
 function towersOfHanoi(startStack, endStack) {
```

Comment:
I was expecting some feedback to the user to let them know if the moves are legal or not.